tests/rkt_image_dependencies_test.go

```diff
@@ -0,0 +1,218 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	taas "github.com/coreos/rkt/tests/test-auth-server/aci"
+	"github.com/coreos/rkt/tests/testutils"
+)
+
+const (
+	manifestDepsTemplate = `
+{
+   "acKind" : "ImageManifest",
+   "acVersion" : "0.7.3",
+   "dependencies" : [
+      DEPENDENCIES
+   ],
+   "labels" : [
+      {
+         "name" : "version",
+         "value" : "VERSION"
+      },
+      {
+         "name" : "arch",
+         "value" : "amd64"
+      },
+      {
+         "value" : "linux",
+         "name" : "os"
+      }
+   ],
+   "app" : {
+      "user" : "0",
+      "exec" : [
+         "/inspect", "--print-msg=HelloDependencies"
+      ],
+      "workingDirectory" : "/",
+      "group" : "0",
+      "environment" : [
+      ]
+   },
+   "name" : "IMG_NAME"
+}
+`
+)
+
+// TestImageRender tests 'rkt image render', it will import some existing empty
+// image with a dependency on an image with the inspect binary, render it with
+// rkt image render and check that the exported image has the /inspect file and
+// that its hash matches the original inspect binary hash
+func TestImageDependencies(t *testing.T) {
+	tmpDir := createTempDirOrPanic("rkt-TestImageDeps-")
+	defer os.RemoveAll(tmpDir)
+
+	ctx := testutils.NewRktRunCtx()
+	defer ctx.Cleanup()
+
+	// We need a http server rather than https because httpsOrHTTP()
+	// fallbacks to http when the certificate is not valid.
+	// https://github.com/appc/spec/blob/master/discovery/http.go#L58
+
+	// Actually, httptest.Start() is broken because it waits forever, so
+	// I have to use httptest.StartTLS().
+	// So, patching appc/spec to use InsecureSkipVerify: true
+	serverURL := flag.Lookup("httptest.serve")
+	serverURL.Value.Set("127.0.0.1:443")
+
+	server, err := taas.NewServerWithPaths(taas.None, 20, "", "")
```

Comment:
You already borrowed some code from `rkt_auth_test` (the `serverHandler` function), so here you can just use the `runServer` function instead.